### PR TITLE
Add option to enable printing query statement in otel attribute db.statement

### PR DESCRIPTION
**Pull request by JJ-WM**
# Description

db.statement are not getting printed in trace according to https://opentelemetry.io/docs/specs/semconv/database/database-spans/ with rational is customer data would be leaked in the query statement.  However, users should still have the option to print the query and sanitize it in their own tracing implementation.  

This change will add a new options to print query statement in db.statement attribute.
`new CosmosClientTelemetryConfig().showQueryStatement();`

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by JJ-WM** [comment]
@microsoft-github-policy-service agree [company=”Walmart”]
@microsoft-github-policy-service agree
@microsoft-github-policy-service agree company="Microsoft"

From: "microsoft-github-policy-service[bot]" ***@***.***>
Reply-To: Azure/azure-sdk-for-java ***@***.***>
Date: Wednesday, May 1, 2024 at 12:08 PM
To: Azure/azure-sdk-for-java ***@***.***>
Cc: JJ Chen ***@***.***>, Mention ***@***.***>
Subject: EXT: Re: [Azure/azure-sdk-for-java] Add option to enable printing query statement in otel attribute db.statement (PR #39989)

EXTERNAL: Report suspicious emails to Email Abuse.

@JJ-WM<https://urldefense.com/v3/__https:/github.com/JJ-WM__;!!IfjTnhH9!TXGFFDfYnqKvUBx3M5qdHpsIv7pA9eSAGCU79bYaDPfKZd4U1bNjT88d4jTdmlIaeuifG3hbAavf8BU29cZK8zyA$> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.

@microsoft-github-policy-service agree [company="{your company}"]

Options:

  *   (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

@microsoft-github-policy-service agree

  *   (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.

@microsoft-github-policy-service agree company="Microsoft"
Contributor License Agreement
Contribution License Agreement

This Contribution License Agreement (“Agreement”) is agreed to by the party signing below (“You”),
and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your
contributions to Microsoft open source projects. This Agreement is effective as of the latest signature
date below.

  1.  Definitions.
“Code” means the computer software code, whether in human-readable or machine-executable form,
that is delivered by You to Microsoft under this Agreement.
“Project” means any of the projects owned or managed by Microsoft and offered under a license
approved by the Open Source Initiative (www.opensource.org<https://urldefense.com/v3/__http:/www.opensource.org__;!!IfjTnhH9!TXGFFDfYnqKvUBx3M5qdHpsIv7pA9eSAGCU79bYaDPfKZd4U1bNjT88d4jTdmlIaeuifG3hbAavf8BU29dn7f13e$>).
“Submit” is the act of uploading, submitting, transmitting, or distributing code or other content to any
Project, including but not limited to communication on electronic mailing lists, source code control
systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
discussing and improving that Project, but excluding communication that is conspicuously marked or
otherwise designated in writing by You as “Not a Submission.”
“Submission” means the Code and any other copyrightable material Submitted by You, including any
associated comments and documentation.
  2.  Your Submission. You must agree to the terms of this Agreement before making a Submission to any
Project. This Agreement covers any and all Submissions that You, now or in the future (except as
described in Section 4 below), Submit to any Project.
  3.  Originality of Work. You represent that each of Your Submissions is entirely Your original work.
Should You wish to Submit materials that are not Your original work, You may Submit them separately
to the Project if You (a) retain all copyright and license information that was in the materials as You
received them, (b) in the description accompanying Your Submission, include the phrase “Submission
containing materials of a third party:” followed by the names of the third party and any licenses or other
restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
guidelines concerning Submissions.
  4.  Your Employer. References to “employer” in this Agreement include Your employer or anyone else
for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
Submission is made in the course of Your work for an employer or Your employer has intellectual
property rights in Your Submission by contract or applicable law, You must secure permission from Your
employer to make the Submission before signing this Agreement. In that case, the term “You” in this
Agreement will refer to You and the employer collectively. If You change employers in the future and
desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
and secure permission from the new employer before Submitting those Submissions.
  5.  Licenses.

  *   Copyright License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
parties.
  *   Patent License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
Your patent claims that are necessarily infringed by the Submission or the combination of the
Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
import or otherwise dispose of the Submission alone or with the Project.
  *   Other Rights Reserved. Each party reserves all rights not expressly granted in this Agreement.
No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
granted by implication, exhaustion, estoppel or otherwise.

  1.  Representations and Warranties. You represent that You are legally entitled to grant the above
licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
have disclosed under Section 3). You represent that You have secured permission from Your employer to
make the Submission in cases where Your Submission is made in the course of Your work for Your
employer or Your employer has intellectual property rights in Your Submission by contract or applicable
law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
  2.  Notice to Microsoft. You agree to notify Microsoft in writing of any facts or circumstances of which
You later become aware that would make Your representations in this Agreement inaccurate in any
respect.
  3.  Information about Submissions. You agree that contributions to Projects and information about
contributions may be maintained indefinitely and disclosed publicly, including Your name and other
information that You submit with Your Submission.
  4.  Governing Law/Jurisdiction. This Agreement is governed by the laws of the State of Washington, and
the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
defenses of lack of personal jurisdiction and forum non-conveniens.
  5.  Entire Agreement/Assignment. This Agreement is the entire agreement between the parties, and
supersedes any and all prior agreements, understandings or communications, written or oral, between
the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

**Comment by JJ-WM** [comment]
@microsoft-github-policy-service agree

and

@microsoft-github-policy-service agree company="Walmart"


From: "microsoft-github-policy-service[bot]" ***@***.***>
Reply-To: Azure/azure-sdk-for-java ***@***.***>
Date: Wednesday, May 1, 2024 at 12:54 PM
To: Azure/azure-sdk-for-java ***@***.***>
Cc: JJ Chen ***@***.***>, Mention ***@***.***>
Subject: EXT: Re: [Azure/azure-sdk-for-java] Add option to enable printing query statement in otel attribute db.statement (PR #39989)

EXTERNAL: Report suspicious emails to Email Abuse.

@JJ-WM<https://urldefense.com/v3/__https:/github.com/JJ-WM__;!!IfjTnhH9!TcnxvigESToGNSY4XSCGSXdup7CX9jWV-ugyHWgjYZL_U_06LTaOkYaCShm_n3Lq_0GrUzkR9_3tU_GVRYbMkIEu$> the command you issued was incorrect. Please try again.

Examples are:

@microsoft-github-policy-service agree

and

@microsoft-github-policy-service agree company="your company"

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java [hunk 1]
```diff
@@ -1116,19 +1113,20 @@ private void newTimeout(
         this.acquisitionAndIdleEndpointDetectionTimeout.set(acquisitionAndIdleEndpointDetectionTimer.newTimeout(
             (Timeout timeout) -> {
                 if (idleEndpointTimeoutInNanos == 0) {
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Idle endpoint check is disabled");
+                    // log is too noisy when debug is enabled
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("Idle endpoint check is disabled");
```
Comment (Gueorgi) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:1118]: why this log level change?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java [hunk 2]
```diff
@@ -1116,19 +1113,20 @@ private void newTimeout(
...
                     }
                 } else {
                     final long elapsedTimeInNanos = System.nanoTime() - endpoint.lastRequestNanoTime();
 
                     if (idleEndpointTimeoutInNanos - elapsedTimeInNanos <= 0) {
-                        if (logger.isDebugEnabled()) {
+                        if(logger.isDebugEnabled()) {
```
Comment (Gueorgi) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:1124]: nit: add back the space after if as it was before for consistency and readablity

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java [hunk 3]
```diff
@@ -1116,19 +1113,20 @@ private void newTimeout(
...
                             logger.debug(
-                                "{} closing endpoint due to inactivity (elapsedTime: {} > idleEndpointTimeout: {})",
-                                endpoint,
-                                Duration.ofNanos(elapsedTimeInNanos),
-                                Duration.ofNanos(idleEndpointTimeoutInNanos));
+                                    "{} closing endpoint due to inactivity (elapsedTime: {} > idleEndpointTimeout: {})",
+                                    endpoint,
+                                    Duration.ofNanos(elapsedTimeInNanos),
+                                    Duration.ofNanos(idleEndpointTimeoutInNanos));
```
Comment (Gueorgi) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:1129]: Perhaps no need to do all these formattings? May be turn off your IDEs settings for that? Every one has different preferences for tabs, spaces etc.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java [hunk 4]
```diff
@@ -1222,18 +1220,15 @@ private void notifyChannelConnect(final ChannelFuture future, final Promise<Chan
                         return channel;
                     });
                 } else {
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("notifyChannelConnect promise.trySuccess(channel)=false");
-                    }
+                    logger.info("notifyChannelConnect local {}, remote {} promise.trySuccess(channel)=false", channel.localAddress(), channel.remoteAddress());
```
Comment (Gueorgi) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:1223]: why change this to info level?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java [hunk 5]
```diff
@@ -1222,18 +1220,15 @@ private void notifyChannelConnect(final ChannelFuture future, final Promise<Chan
...
 
                     // Promise was completed in the meantime (like cancelled), just close the channel
                     this.closeChannel(channel);
                 }
 
             } else {
-                if (logger.isDebugEnabled()) {
-                    logger.debug("notifyChannelConnect future was not successful");
-                }
+                logger.error("notifyChannelConnect future was not successful {}", future.cause());
```
Comment (Gueorgi) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:1230]: do we need to log this as error if it was not logged before as such? how often this happens and is it logged in some other way?

**Comment by Gueorgi** [comment]
@JJ-WM are there files missing from the PR? It has single file changed and no change to CosmosClientTelemetryConfig looks like.

**Pull request closed without merging**